github-actions/google-internal-tests/lib/main.ts

```diff
@@ -0,0 +1,79 @@
+import core from '@actions/core';
+import {context} from '@actions/github';
+import {Octokit} from '@octokit/rest';
+import minimatch from 'minimatch';
+
+async function main() {
+  if (context.repo.owner !== 'angular') {
+    core.info('Skipping Google Internal Tests action for non-Angular repos.');
+    return;
+  }
+  if (context.eventName !== 'pull_request_target' || context.payload.action !== 'synchronize') {
+    throw new Error(
+      'Expected Google Internal Tests action to be triggered for ' +
+        '`pull_request_target.synchronize` events.',
+    );
+  }
+
+  const githubToken = core.getInput('github-token', {required: true});
```

Comment:
Can we use the angular robot instead of the Github Actions token?

Reply:
Resolved offline. Using the github action token we can limit permissions better